packages/app/components/Queue/QueueCard.tsx

```diff
@@ -31,47 +54,121 @@ const CenterRow = styled(Row)`
 `;
 
 interface QueueCardProps {
+  role: Role;
+  helping: boolean;
   rank: number;
   name: string;
   questionType: QuestionType;
   waitTime: number;
   status: string;
+  onOpen: (name: string) => void;
 }
 
 export default function QueueCard({
+  role,
+  helping,
   rank,
   name,
   questionType,
   waitTime,
   status,
+  onOpen,
 }: QueueCardProps) {
   return (
-    <HorizontalCard>
-      <CenterRow justify="space-between">
-        <Col span={1}>
-          <Text>{rank}</Text>
-        </Col>
-        <Col xs={16} sm={11} lg={6}>
-          <CenterRow>
-            <Avatar icon={<UserOutlined />} />
-            <NameContainer>
-              <Text>{name}</Text>
-            </NameContainer>
+    <div>
+      {role === "ta" && !helping && (
```

Comment:
for the `"ta"` and `"student"`, could we make `Role` an `enum` instead of union string to have more consistent typing?